platforms/joystick/joystick_driver.go

```diff
@@ -57,20 +59,12 @@ type pair struct {
 	ID   int    `json:"id"`
 }
 
-// hat is a JSON representation of hat, name and id
```

Comment:
Are the hat-switches now part of the buttons or are they no longer supported?

Reply:
Confirmed they are buttons now.

Reply:
CORRECTION: dualshock4 on linux, they are joysticks. dualshock4 on macOS apparently not supported.